uxarray/grid/coordinates.py

```diff
@@ -235,3 +234,88 @@ def _populate_lonlat_coord(grid):
             "long_name": "latitude of mesh nodes",
             "units": "degrees_north",
         })
+
+
+def _populate_centroid_coord(self, repopulate=False):
```

Comment:
Please add a proper description of the limitations of this function and adequate warning. As far as I can tell, this approach can only be applied to some convex polygon (ideally triangle). An exception should be raised if the input polygons are not within the criteria. 

Also, please include the definition of centroid here.

Reply:
Thanks for the review Hongyu! It is my understanding this would work with any convex polygon. Since our definition here is an average of vertices, any convex polygon would be able to be used, if I understand it correctly @philipc2

Reply:
That's what I was talking about. If we can only confirm this will only be valid within a convex polygon, we should include that in the description. And if the input polygon is not convex, a warning should be raised. Otherwise, someone will get a silence crash when they're trying to apply this function on a concave polygon, which we don't want to see happening.

Another thing that brought up my concern here is that we are working on the spherical surface. So my questions here are:

1. The coordinates averaging works in the 3-dimensional space for "planes" like n-dimensional simplex. Does it work for the spherical surface?

2. Even though we assume it works, we need to add the normalization step for your centroid calculation function (I also added a comment there). After we project it to the spherical surface using this naive method, How can we confirm that the projected centroid (if we assume it's the centroid we want) is still within the polygon on the sphere? Because the edge might arc up on the sphere

Reply:
> that's what I was talking about. If we can only confirm this will only be valid within a convex polygon

@paullric @rljacob @erogluorhan 

Do we have any sample mesh files that contain concave faces? I'm interested in how often we may encounter concave faces when working with, for example, grids from climate models, since that's our most applicable use case.